pkg/skaffold/build/deps.go

```diff
@@ -0,0 +1,114 @@
+package build
+
+import (
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/skaffold/pkg/skaffold/config"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type DependencyMap struct {
+	artifacts       []*config.Artifact
+	pathToArtifacts map[string][]*config.Artifact
+}
+
+type DependencyResolver interface {
+	GetDependencies(a *config.Artifact) ([]string, error)
+}
+
+//TODO(@r2d4): Figure out best UX to support configuring this blacklist
+var ignoredPrefixes = []string{"vendor", ".git"}
```

Comment:
Why are we ignoring vendor? That seems wrong...

Reply:
This is for watch. We can't really watch vendor because we keep hitting watch limits